packages/core/src/state-management/state-manager.ts

```diff
@@ -340,14 +346,57 @@ export class StateManager {
     return this._processAnchorResponse(state$, anchorStatus$)
   }
 
-  private async _saveAnchorRequestForState(state$: RunningState): Promise<void> {
+  private async _buildAnchorRequestCARFile(streamId: StreamID, tip: CID): Promise<CAR> {
+    const car = this.carFactory.build()
+
+    // Root block
+    const timestampISO = new Date().toISOString()
+    car.put(
+      {
+        timestamp: timestampISO,
+        streamId: streamId.bytes,
+        tip: tip.bytes,
```

Comment:
```suggestion
        tip: tip,
```

Please just use the CID directly here. IPLD does support putting CIDs directly, I thought I made this clear in a previous comment? 

The neat thing here is that (at least in ipfs) you can do things like `ipfs.dag.get(rootCid, { path: '/tip/link/id' }` to traverse the dag (and in this example case retrieve the genesis commit).

Reply:
Yeah, you had already mentioned that we should have stored CID's directly, it slipped my mind, sorry.